File: Algorithms/Sorting/Python/bubble_sort.py
# Bubble Sort
import random


def populate(arr, items):
    for i in range(items):
        arr.append(random.randint(1,items*100))
    return arr


def bubbleSort(arr):
    for i in range(len(arr)):
        swapped = False
        for j in range(len(arr) - 1):
            if (arr[j] > arr[j + 1]):
                temp = arr[j]
                arr[j] = arr[j + 1]
                arr[j + 1] = temp
                swapped = True
        if swapped == False:
            return arr


if __name__ == '__main__':
    arr = []
    items = 10
    populate(arr, items)
    print('Unsorted: ', arr)
    bubbleSort(arr)
    print('Sorted: ', arr)

File: bubble-sort-java/BubbleSort.java
class BubbleSort {
    public static void main(String[] args) {
    }

    static void bubble(int arr[], int i, int n) {
        boolean isSwapping = false;
        do {
            isSwapping = false;
            for (i = 0; i < arr.length-1; i++) {

                if (arr[i] > arr[i + 1]) {

                    int temp = arr[i];

                    arr[i] = arr[i + 1];

                    arr[i + 1] = temp;

                    isSwapping = true;
                }
            }
        } while (isSwapping);
    }
}
